Sources/Ink/Internal/URLDeclaration.swift

```diff
@@ -7,14 +7,29 @@
 internal struct URLDeclaration: Readable {
     var name: String
     var url: URL
+    var title: Substring?
 
     static func read(using reader: inout Reader) throws -> Self {
         try reader.read("[")
         let name = try reader.read(until: "]")
         try reader.read(":")
         try reader.readWhitespaces()
-        let url = reader.readUntilEndOfLine()
 
-        return URLDeclaration(name: name.lowercased(), url: url)
+        var titleText: Substring? = nil
+        let url = try reader.readCharacters(matching: \.isSameLineNonWhitespace)
+
+        if !reader.didReachEnd {
+			if reader.currentCharacter.isNewline {
+				reader.advanceIndex()
+			}
+			if reader.currentCharacter.isSameLineWhitespace {
+				try reader.readWhitespaces()
+			}
+            if let delimeter = TitleDelimeter(rawValue: reader.currentCharacter) {
+                reader.advanceIndex()
+                titleText = try reader.read(until: delimeter.closing)
```

Comment:
```suggestion
                titleText = try reader.read(until: delimeter.closing)
                reader.discardWhitespaces()
                if !reader.didReachEnd && !reader.currentCharacter.isWhitespace {
                    reader.moveToIndex(index)
                    titleText = nil
                }
```

This helps fix the new failure of CommonMark test 179:
"No further non-whitespace characters may occur on the line." ([spec](https://spec.commonmark.org/0.29/#link-reference-definition) & [example](https://spec.commonmark.org/0.29/#example-179))